### ISSUE TITLE
ui_submit and class name as argument

### ISSUE THREAD

**Issue by maanas** [webmin/webmin#408]
We are working with authentic-theme which is bootstrap based theme. It is noticed that the ui_submit function accept html tags for the last argument but no separate class argument. The class for the submit button is arrived after doing a reverse lookup on get_button_style subroutine. 
It would have been so much better if we can pass class arguments right in the ui_submit function. That will allow better flexibility in theming the whole layout. ui_submit and ui_button are special cases where a button can occur in more than one color.

Another instance is using js to style the button using javascript in the firewall module.
```
authentic theme/unauthenticated/js

$('input[type="submit"]').each(function() {
   $(this).addClass("btn btn-default")
});
```

It appears, all such deviation is required because the ui_submit do not have the ability to pass class arguments. If my presumption is correct then can be update ui_submit function to include a class argument which will simplify the process.

It above proposal is acceptable then We are willing to undertake changes and commit back. The last options can be class option which can be option making it backward compatible.

**Comment by qooob** [comment]
>It would have been so much better if we can pass class arguments right in the ui_submit function.

What are you working on? Whatever, `ui_submit()` can be re-assigned using Perl side on the `theme.pl` file by doing `sub theme_ui_submit {}`. You can build it anyway you want.

**Comment by jcameron** [comment]
+1 to @iliaross's solution

That said, I would be OK with accepting a change to add a class parameter.

**Comment by maanas** [comment]
@iliaross @jcameron We have already implemented the solution you are suggesting. @mvarian has made a pull request https://github.com/iliaross/authentic-theme/pull/520. 
Between both of us (@mvarian & @maanas) were split on choice of option.If you see from a theme perspective, it is a good solution. If you see from a perl perspective, it is more intuitive to add the class right in the ui_submit which can be used across the theme.
Both of you guys are doing awesome work, and we though we bounce this here and let pros take a call.

**Comment by qooob** [comment]
It would only make sense, if all other modules used that function the way you're saying but they are not. Even though I made it the right way, it would be pointless, as non of the modules doing it.

**Comment by mvarian** [comment]
@iliaross For context, much of the work we are doing right now involves custom modules mixing with the standard ones.  We're trying to achieve granular control over their styling while making as few changes as possible to themes, perl, or standard convention.  This is what spurred our conversation about the class parameter to `ui_submit()`.

@maanas Thinking more about this over the weekend, what do you think about passing tags through the current version of `ui_submit()`, and then overriding `theme_ui_submit()` for our specific implementation to detect those tags and act accordingly?  Instead of sending the classes, we send a tag for the type of button.  We could segregate our custom logic by having the tags trigger a custom function.  Would this allow us to do what we want for roughly the same amount of effort or am I overlooking something?

**Comment by qooob** [comment]
Yes, you could use `tags` actually but it would make you to use custom css for buttons or anything you need. You could pass a tag to `ui_submit` _data-button-success_ and on CSS side of your module you could do:

```CSS
button[data-button-success] {
}
button[data-button-success]:hover {
}
button[data-button-success]:active {
}
```

That would work. ;)
